tw-tasks-core-test-spring-boot-starter/src/main/java/com/transferwise/tasks/test/TwTasksCoreTestAutoConfiguration.java

```diff
@@ -0,0 +1,39 @@
+package com.transferwise.tasks.test;
+
+import com.transferwise.tasks.TasksProperties;
+import com.transferwise.tasks.core.autoconfigure.ITwTasksDataSourceProvider;
+import com.transferwise.tasks.dao.ITaskDaoDataSerializer;
+import com.transferwise.tasks.test.dao.ITestTaskDao;
+import com.transferwise.tasks.test.dao.MySqlTestTaskDao;
+import com.transferwise.tasks.test.dao.PostgresTestTaskDao;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
+
+@Configuration
+public class TwTasksCoreTestAutoConfiguration {
+
+  @Bean
+  @ConditionalOnMissingBean(ITestTaskDao.class)
+  @ConditionalOnProperty(value = "tw-tasks.core.db-type", havingValue = "POSTGRES")
+  public PostgresTestTaskDao twTasksPostgresTestTaskDao(ITwTasksDataSourceProvider twTasksDataSourceProvider, TasksProperties properties,
+      ITaskDaoDataSerializer taskDaoDataSerializer) {
+    return new PostgresTestTaskDao(twTasksDataSourceProvider.getDataSource(), properties, taskDaoDataSerializer);
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(ITestTaskDao.class)
+  @ConditionalOnProperty(value = "tw-tasks.core.db-type", havingValue = "MYSQL")
+  public MySqlTestTaskDao twTasksMysqlTestTaskDao(ITwTasksDataSourceProvider twTasksDataSourceProvider, TasksProperties properties,
+      ITaskDaoDataSerializer taskDaoDataSerializer) {
+    return new MySqlTestTaskDao(twTasksDataSourceProvider.getDataSource(), properties, taskDaoDataSerializer);
+  }
+
+  @Bean
+  @Primary()
```

Comment:
Not sure if we need this to be primary. If I add this dependency I can probably remove local configuration of `TestTasksService` as well as I probably understand why I'm adding the dependency 🤔 Otherwise we end up having code in the service that configures `TestTasksService` instance which is not used

Reply:
Good question, the idea was to make sure TasksService gets overwritten by this.

But there is also a condition.
@ConditionalOnMissingBean(ITasksService.class)
public TasksService twTasksTasksService()

I'll test it out in another PR. This one at least worked locally in payout service.